### PR TITLE
Replacing const char* arguments with std::string in error functions

### PR DESCRIPTION
**Pull request by syclik**
#### Submission Checklist

- [x] Run unit tests: `./runTests.py test/unit`
- [x] Run cpplint: `make cpplint`
- [x] Declare copyright holder and open-source license: see below

#### Summary:
Cleaning up code by replacing uses of `const char*` with `const std::string&`.

This shouldn't affect the behavior of the code.

#### Intended Effect:
Make our code base more idiomatic C++ and more consistent.

#### How to Verify:
Check the diff. Most of it is going to be very repetitive.

#### Side Effects:
None.

#### Documentation:
None.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

Stan Group

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: stan/math/prim/scal/prob/normal_sufficient_lpdf.hpp [hunk 1]
```diff
@@ -15,13 +14,14 @@
...
 #include <stan/math/prim/scal/meta/max_size.hpp>
+#include <string>
 
 namespace stan {
 
   namespace math {
 
     /**
-     * The log of the normal density for the specified scalar(s) given
+     * The log of the normal density for the specified scalar(s given
```
Comment (seantalts) [inline stan/math/prim/scal/prob/normal_sufficient_lpdf.hpp]: probably mistakenly deleted the `)` here

File: stan/math/prim/scal/prob/wiener_lpdf.hpp [hunk 2]
```diff
@@ -73,7 +73,7 @@ namespace stan {
     typename return_type<T_y, T_alpha, T_tau, T_beta, T_delta>::type
     wiener_lpdf(const T_y& y, const T_alpha& alpha, const T_tau& tau,
                const T_beta& beta, const T_delta& delta) {
-      static const char* function("wiener_lpdf(%1%)");
+      static const std::string function = "wiener_lpdf";
```
Comment (seantalts) [inline stan/math/prim/scal/prob/wiener_lpdf.hpp]: Was this a mistake before and it's fixed now? I see some other `(%1%)` in the code; are they used as sort of format strings?

**Comment by bob-carpenter** [comment]
@seantalts  Yes, we definitely want to make this change.

I'm pretty sure all those `%1%` things are holdovers from back when we were using boost formatting that just got code transformed along with everything else.  We stopped using Boost formatting because it has the same awful runtime error behavior as `printf`.

File: stan/math/prim/scal/prob/normal_sufficient_lpdf.hpp (hunk 1 above)
Comment (syclik) [inline stan/math/prim/scal/prob/normal_sufficient_lpdf.hpp]: Yes. Fixing. (I'll walk through all the changes to make sure I didn't mistakenly delete anything else.)

**Comment by syclik** [comment]
+1 to Bob's comment.

The `%1%` was from when we used boost's formatting for output messages (when we had boost's policies in place too). I thought we had removed it, but I found a dangling instance, so cleaned it up in the process.

**Comment by syclik** [comment]
@seantalts thanks for picking up on that accidentally deleted paren. Fixed, pushed. Waiting on tests now.

**Pull request merged**